src/App.js

```diff
@@ -66,51 +81,44 @@ function saveDeviceData() {
 }
 
 function App() {
+  const [session, setSession] = useState(null); // Stores the Session
   const incodeContainerRef = useRef(null);
-  const [scriptLoaded, setScriptLoaded] = useState(false);
-
+  
+  // Store data that will not trigger re-renders unless specifically told so
+  const isLoaded = useRef(false); 
+  
+  // Run this after the initial loading
   useEffect(() => {
-    if (!SDKActive) {
-      SDKActive = true;
-
-      const script = document.createElement("script");
-      script.src = process.env.REACT_APP_INCODE_WEB_SDK_URL;
-      script.async = true;
-
-      script.onload = async () => {
-        console.log("Incode Web SDK Loaded");
-        setScriptLoaded(true);
-        incode = window.OnBoarding.create({
-          apiURL: process.env.REACT_APP_INCODE_API_URL,
-          translations
-        });
-        incodeSession = await getIncodeToken();
-
-        container = incodeContainerRef.current;
-
-        saveDeviceData();
-        captureIdFrontSide();
-      };
-      script.onerror = () => {
-        console.error("Error loading the Incode Web SDK");
-        setScriptLoaded(false);
-      };
-
-      document.body.appendChild(script);
-
-      return () => {
-        document.body.removeChild(script);
-      };
-    }
+    // Only fetch the data if we havent fetched it yet
+    if (isLoaded.current) return;
+    
+    //Fetch the session and save it on the session variable
+    startOnboardingSession().then(async (session) => {
+      setSession(session);
+    }).catch(
+      (e)=>console.log(e)
+    );
+    
+    // We already sent the async call, don't call it again
+    isLoaded.current = true;
   }, []);
-
+  
+  useEffect(()=>{
```

Comment:
this gets called after setSession is resolved at the end of the async fetch above